doc/gdb-debugging.md

```diff
@@ -0,0 +1,85 @@
+### GDB Debugging
+This document describes the options available for debugging node using the GNU debugger.
+
+We don't provide the debuginfo package, which contains the symbols required for debugging, as this would make our runtime image about 480MB larger. On systems where root access is allowed then the debuinfo package can simply be installed using yum/rpm, but on others like OpenShift getting a pod to run with root access can be somewhat tricky. 
+
+### OpenShift
+When running an application on OpenShift the pod will not have root access. It is possible to configure such access but that means configuring the OpenShift environment. This section provides an alternative to that which allows debugging the application using a remote gdb session. This is done by starting a container in the same cluster with an image with the node runtime and debuginfo, and then connecting to the pod that is the target of the debugging.
+
+The example used in this document is [nodejs-rest-http](https://github.com/bucharest-gold/nodejs-rest-http).
+
+#### Start/attach to the running process
+First step is to start the gdbserver on the pod that is the target of the debugging session:
+```
+$ oc rsh nodejs-rest-http-1-nw7hz
+```
+We want to attach the `gdbserver` to the node process (`30` in this case but can differ):
+```console
+sh-4.2$ ps -ef
+UID        PID  PPID  C STIME TTY          TIME CMD
+1000120+     1     0  0 03:58 ?        00:00:00 npm
+1000120+    30     1  0 03:58 ?        00:00:00 node .
+1000120+    41     0  0 03:59 ?        00:00:00 /bin/sh
+1000120+    45    41  0 03:59 ?        00:00:00 ps -ef
+$ gdbserver 172.17.0.6:7777 --attach 30
+Attached; pid = 30
+Listening on port 7777
+```
+The gdbserver is now listening for connections.
+
+#### Start the debuginfo image
+The next step is to start a container with the debuginfo image in the same cluster.
+
+```console
+$ oc run -i -t nodejs-debuginfo --image=bucharestgold/centos7-s2i-nodejs-debuginfo:10.x --restart=Never
+```
+Running the above command will drop you into the newly created container where 
+the gdb session can be started. This is done using the following command: 
+```console
+bash-4.2$ gdb node
+Reading symbols from /usr/bin/node...Reading symbols from /usr/lib/debug/usr/bin/node.debug...done.
+done.
+(gdb)
+```
+Then, start a remote debugging session to the target pod:
+```console
+(gdb) target remote 172.17.0.6:7777
```

Comment:
I get this when running the "target remote" command
```
(gdb) target remote 172.17.0.8:7777
Remote debugging using 172.17.0.8:7777
warning: Could not load shared library symbols for /lib64/libnss_wrapper.so.
Do you need "set solib-search-path" or "set sysroot"?
Reading symbols from /lib64/libdl.so.2...(no debugging symbols found)...done.
Loaded symbols for /lib64/libdl.so.2
Reading symbols from /lib64/librt.so.1...(no debugging symbols found)...done.
Loaded symbols for /lib64/librt.so.1
Reading symbols from /lib64/libstdc++.so.6...(no debugging symbols found)...done.
Loaded symbols for /lib64/libstdc++.so.6
Reading symbols from /lib64/libm.so.6...(no debugging symbols found)...done.
Loaded symbols for /lib64/libm.so.6
Reading symbols from /lib64/libgcc_s.so.1...(no debugging symbols found)...done.
Loaded symbols for /lib64/libgcc_s.so.1
Reading symbols from /lib64/libpthread.so.0...(no debugging symbols found)...done.
Loaded symbols for /lib64/libpthread.so.0
Reading symbols from /lib64/libc.so.6...(no debugging symbols found)...done.
Loaded symbols for /lib64/libc.so.6
Reading symbols from /lib64/ld-linux-x86-64.so.2...(no debugging symbols found)...done.
Loaded symbols for /lib64/ld-linux-x86-64.so.2
0x00007ffbc81f8ec9 in syscall () from /lib64/libc.so.6
Missing separate debuginfos, use: debuginfo-install glibc-2.17-222.el7.x86_64 libgcc-4.8.5-28.el7_5.1.x86_64 libstdc++-4.8.5-28.el7_5.1.x86_64
```

Reply:
This is because we don't include debuginfo for glibc, libgcc etc. I tried adding them but that made the image about 2 GB which took too long to load. Plus I think it is enough to be able to debug node source code.

Reply:
Would it be worthwhile to mention in the doc that users may see this message so they are not confused?

Reply:
No problem,  i posted this since i thought it was an error.  but then i actually tried the next steps, and those worked.  So "nothing to see here" :)

Reply:
I'll see if those can be silenced in some way...but otherwise I'll explain it in the doc.

Reply:
I've made an update and the output of running `gdb node` should now be:
```console
bash-4.2$ gdb node
Reading symbols from /usr/bin/node...Reading symbols from /usr/lib/debug/usr/bin/node.debug...done.
done.
(gdb)
```